mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/DynamicConfigDeclarationEnricher.java

```diff
@@ -6,35 +6,41 @@
  */
 package org.mule.runtime.extension.internal.loader.enricher;
 
+import static java.util.Optional.empty;
 import static org.mule.runtime.extension.api.loader.DeclarationEnricherPhase.STRUCTURE;
 import static org.mule.runtime.extension.api.util.ExtensionModelUtils.acceptsExpressions;
 import static org.mule.runtime.extension.internal.loader.util.InfrastructureParameterBuilder.addExpirationPolicy;
+
 import org.mule.runtime.api.meta.model.declaration.fluent.ConfigurationDeclaration;
 import org.mule.runtime.api.meta.model.declaration.fluent.ParameterizedDeclaration;
-import org.mule.runtime.extension.api.loader.DeclarationEnricher;
 import org.mule.runtime.extension.api.loader.DeclarationEnricherPhase;
 import org.mule.runtime.extension.api.loader.ExtensionLoadingContext;
+import org.mule.runtime.extension.api.loader.WalkingDeclarationEnricher;
 import org.mule.runtime.extension.api.runtime.ExpirationPolicy;
 
+import java.util.Optional;
+
 /**
  * Adds an {@link ExpirationPolicy} parameter to all configs which might be used in a dynamic way
  *
  * @since 1.0
  */
-public class DynamicConfigDeclarationEnricher implements DeclarationEnricher {
+public class DynamicConfigDeclarationEnricher implements WalkingDeclarationEnricher {
 
   @Override
   public DeclarationEnricherPhase getExecutionPhase() {
     return STRUCTURE;
   }
 
   @Override
-  public void enrich(ExtensionLoadingContext extensionLoadingContext) {
+  public Optional<DeclarationEnricherWalkDelegate> getWalkDelegate(ExtensionLoadingContext extensionLoadingContext) {
```

Comment:
this one should always return a delegate that has an onConfigurationMethod that adds the expiration policy, or be a normal DeclarationEnricher.